packages/databricks-vscode/src/file-managers/DatabricksYamlFile.ts

```diff
@@ -0,0 +1,99 @@
+import path from "path";
+import {RemoteUri, LocalUri} from "../sync/SyncDestination";
+import {FileUtils} from "../utils";
+import {WorkspaceStateManager} from "../vscode-objs/WorkspaceState";
+import * as fs from "node:fs/promises";
+import YAML from "yaml";
+
+/* eslint-disable @typescript-eslint/naming-convention */
+interface DatabricksYaml {
+    environments: {
+        [key: string]: {
+            compute_id?: string;
+            workspace: {
+                host: string;
+                root_path?: string;
+            };
+            mode: "development";
+        };
+    };
+}
+/* eslint-enable @typescript-eslint/naming-convention */
+
+export class DatabricksYamlFile {
+    constructor(
+        readonly workspaceUrl: URL,
+        public clusterId?: string,
+        public workspacePath?: RemoteUri
+    ) {}
+
+    private toYaml(workspaceState: WorkspaceStateManager) {
+        const environments: DatabricksYaml["environments"] = {};
+        /* eslint-disable @typescript-eslint/naming-convention */
+        environments[DatabricksYamlFile.getTargetName(workspaceState)] = {
+            compute_id: this.clusterId,
+            workspace: {
+                host: this.workspaceUrl.toString(),
+                root_path: this.workspacePath?.path,
+            },
+            mode: "development",
+        };
+        /* eslint-enable @typescript-eslint/naming-convention */
+
+        const doc = new YAML.Document({environments});
+        return doc.toString();
+    }
+
+    static getTargetName(workspaceState: WorkspaceStateManager) {
+        return `databricks-vscode-${workspaceState.fixedUUID.slice(0, 8)}`;
```

Comment:
what is the consideration between adding a new environment or simply overriding values in the root of the config?